Editor/TestTools/EditorImguiTester/EditorImguiTest.cs

```diff
@@ -103,36 +109,35 @@ public IEnumerator Test()
             try
             {
                 result = BaseGiven();
-
-                BaseWhen(result, (sender, args) =>
-                {
-                    try
-                    {
-                        BaseThen(args.Result);
-                        state = TestState.Passed;
-                    }
-                    catch (Exception e)
-                    {
-                        Debug.LogErrorFormat("Test {0} failed: {1}", GetType().GetNameWithNesting(), e.Message);
-                        state = TestState.Failed;
-                    }
-                });
             }
             catch (Exception e)
             {
-                Debug.LogErrorFormat("Test {0} failed: {1}", GetType().GetNameWithNesting(), e.Message);
+                Debug.LogErrorFormat("Test {0} failed during initialization: {1}", GetType().GetNameWithNesting(), e);
                 state = TestState.Failed;
             }
 
+            //Make sure that we call when/then outside of the OnGUI call so RepaintImmediately would not throw an error.
+            EditorApplication.delayCall += () =>
+            {
+                try
+                {
+                    BaseWhen(result);
+                    BaseThen(result);
```

Comment:
What do you think about enabling `BaseThen` to be yielded? That could allow more complex assertions. 🤔

Reply:
Could you give me example? I can't think of anything that would justify the complexity. Also, we can always do it later.

Reply:
Agree, it does not have to be done now.

But for example, there could be a case where we need to wait until something it is applied (perhaps a frame), yielding the BaseThen would allow us to yield null or `WaitUntil`.

A potential scenario: a test where a behavior or condition that requires a property is just added to a step, then a new game object is added as a parameter and the "Fix Button" is pressed. We might want to wait for a frame to ensure the required components were added.